src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java

```diff
@@ -407,10 +387,21 @@ private static void executeDelete(String index, String type, String id, Client e
 		
 		logger.debug("DELETE: '{}/{}/{}'", index, type, id);
 		
+		int timeout = 30,
+				count = 0;
+		
 		DeleteRequest request = esClient.prepareDelete(index, type, id).request();
 		try {
-			esClient.delete(request).get();
-		} catch (InterruptedException | ExecutionException e) {
+			ActionFuture<DeleteResponse> response = esClient.delete(request);
+			// wait until response isDone()
+			while ( !response.isDone() ) {
```

Comment:
I think http://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html#get-long-java.util.concurrent.TimeUnit- is what you desire instead of Thread.sleep